plugins/polio/api/vaccines/stock_management.py

```diff
@@ -78,11 +75,169 @@ class VaccineStockManagementReadWritePerm(GenericReadWritePerm):
+    search_fields = ["vaccine", "country__name"]
+
+    @action(detail=True, methods=["get"])
+    def usable_vials(self, request, pk=None):
+        if pk is None:
+            return Response({"error": "No VaccineStock ID provided"}, status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            vaccine_stock = self.get_queryset().get(id=pk)
+        except VaccineStock.DoesNotExist:
+            return Response({"error": "VaccineStock not found"}, status=status.HTTP_404_NOT_FOUND)
+
+        # First find the corresponding VaccineRequestForms
+        vrfs = VaccineRequestForm.objects.filter(
+            campaign__country=vaccine_stock.country, vaccine_type=vaccine_stock.vaccine
+        )
+        if not vrfs.exists():
+            return Response(
+                {"error": "No VaccineRequestForm found for the given VaccineStock Country and Vaccine Type"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        # Then find the corresponding VaccineArrivalReports
+        arrival_reports = VaccineArrivalReport.objects.filter(request_form__in=vrfs)
+        if not arrival_reports.exists():
+            return Response(
+                {"error": "No VaccineArrivalReport found for the given VaccineStock Country and Vaccine Type"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        # Use the doses_per_vial from the latest VaccineArrivalReport for all calculations
+        latest_arrival_report = arrival_reports.latest("arrival_report_date")
+
+        destruction_reports = DestructionReport.objects.filter(vaccine_stock=vaccine_stock).order_by(
+            "destruction_report_date"
+        )
+        incident_reports = IncidentReport.objects.filter(vaccine_stock=vaccine_stock).order_by(
+            "date_of_incident_report"
+        )
+        stock_movements = OutgoingStockMovement.objects.filter(vaccine_stock=vaccine_stock).order_by("report_date")
+
+        results = []
+
+        for report in arrival_reports:
+            results.append(
+                {
+                    "date": report.arrival_report_date,
+                    "action": "PO #" + report.po_number if report.po_number else "Stock Arrival",
+                    "vials_in": report.vials_received,
+                    "doses_in": report.doses_received,
+                    "vials_out": None,
+                    "doses_out": None,
+                }
+            )
+
+        for report in destruction_reports:
+            results.append(
+                {
+                    "date": report.destruction_report_date,
+                    "action": f"{report.action} ({report.lot_number})"
+                    if len(report.action) > 0
+                    else f"Stock Destruction ({report.lot_number})",
+                    "vials_in": None,
+                    "doses_in": None,
+                    "vials_out": report.unusable_vials_destroyed,
+                    "doses_out": "N/A",
```

Comment:
This is because at destruction stage, we don't know what type of vial they destroy, so we can't calculate how it translates in doses?
If so maybe add a comment